### PR TITLE
BDEF: Behavior Definition

### PR DESCRIPTION
**Pull request by ThomasPloski**
Pull Request #2021

### CODE REVIEW

**Comment by larshp** [comment]
new `is_active` method added in https://github.com/larshp/abapGit/pull/2017, guess it is also needed as part of this PR

File: src/objects/zcl_abapgit_object_bdef.clas.abap [hunk 1]
```diff
@@ -0,0 +1,457 @@
...
+          <lv_objtype_tr> = ms_item-obj_type.
+          ASSIGN COMPONENT 'SUBTYPE_WB' OF STRUCTURE <ls_object_type_tmp> TO <lv_subtype_wb>.
+          <lv_subtype_wb> = 'BDO'.
+
+          " Remapping to API data structure due to backward compatibility to NW 7.02
+          CREATE DATA lr_object_name_tmp TYPE ('SEU_OBJKEY').
+          ASSIGN lr_object_name_tmp->* TO <lv_object_name_tmp>.
+          <lv_object_name_tmp> = ms_item-obj_name.
+
+          CALL METHOD ('zcl_WB_OBJECT_OPERATOR_FACTORY')=>('CREATE_OBJECT_OPERATOR')
```
Comment (larshp) [inline src/objects/zcl_abapgit_object_bdef.clas.abap]: **zcl**_WB_OBJECT_OPERATOR_FACTORY

File: src/objects/zcl_abapgit_object_bdef.clas.abap [hunk 2]
```diff
@@ -0,0 +1,457 @@
...
```
Comment (larshp) [inline src/objects/zcl_abapgit_object_bdef.clas.abap]: looks like the call to
`cl_WB_OBJECT_OPERATOR_FACTORY')=>('CREATE_OBJECT_OPERATOR'`
is identical in 5 places, can this be moved to private method?

File: src/objects/zcl_abapgit_object_bdef.clas.abap [hunk 3]
```diff
@@ -0,0 +1,457 @@
...
+          <lv_objtype_tr> = ms_item-obj_type.
+          ASSIGN COMPONENT 'SUBTYPE_WB' OF STRUCTURE <ls_object_type_tmp> TO <lv_subtype_wb>.
+          <lv_subtype_wb> = 'BDO'.
+
+          " Remapping to API data structure due to backward compatibility to NW 7.02
+          CREATE DATA lr_object_name_tmp TYPE ('SEU_OBJKEY').
+          ASSIGN lr_object_name_tmp->* TO <lv_object_name_tmp>.
+          <lv_object_name_tmp> = ms_item-obj_name.
+
+          DATA(obj_operator) =  cl_wb_object_operator_factory=>create_object_operator(
```
Comment (larshp) [inline src/objects/zcl_abapgit_object_bdef.clas.abap]: downport missing

**Comment by larshp** [comment]
Also please open a PR to https://github.com/abapGit-tests/BDEF with a simple example of the serialized object

**Comment by absap** [comment]
closing it for now and recreate later with cleaner version

**Comment by ThomasPloski** [comment]
Coding will be adapted accordingly in a seperate PR. Therefore I close this one.

**Pull request closed without merging**